### PR TITLE
gh-122085: Use include files for `whatsnew/3.13.rst` deprecations

### PR DESCRIPTION
**Pull request by hugovk**
Re: https://discuss.python.org/t/streamline-whats-new-by-moving-deprecations-and-removals-out-of-news/53997/8

This moves these sections out of "What's New in Python 3.13" into include files:

* "Pending Removal in Python 3.14"
* "Pending Removal in Python 3.15"
* "Pending Removal in Future Versions"

Later PRs will add includes to other What's New files, to avoid needing to duplicate and sync these sections between files. And we can later also create a dedicated page to list deprecations.

----
📚 Documentation preview 📚: https://cpython-previews--121241.org.readthedocs.build/

### CODE REVIEW

**Comment by nineteendo** [comment]
What happens if an entry from "Pending Removal in Future Versions" moves to "Pending Removal in Python 3.16"?

**Comment by hugovk** [comment]
Then we move it from `../deprecations/pending-removal-in-future.rst` into a `.. include:: ../deprecations/pending-removal-in-3.16.rst`.

**Comment by nineteendo** [comment]
Yes, but then it wouldn't show up in the list of future deprecations for 3.13, 3.16 isn't included (which would be easy to forget when added).

**Comment by hugovk** [comment]
Correct, when creating a new include file, you need to remember to add it to the relevant pages.

**Review by willingc — APPROVED** [review]

**Comment by encukou** [comment]
This only covers the C API, not the Python API. Is that intended?

**Comment by hugovk** [comment]
TODO: This was meant to be for the Python deprecations, not C API ones. I'll update it.

**Comment by nineteendo** [comment]
What will we do with the C API? Do we include it like [this](https://stackoverflow.com/questions/54518160/python-sphinx-include-directive-ignore-the-header-from-included-file)? Or do we put it in a separate file?

```rst
.. include::  ../deprecations/pending-removal-in-3.14.rst
    :start-after: .. start_pending_removal_in_3_14_capi
    :end-before: end_pending_removal_in_3_14_capi
```

**Comment by hugovk** [comment]
No, I think it would be easier to have each in their own files (not necessarily named like this):

```
deprecations/pending-removal-in-3.14.rst
deprecations/pending-removal-in-3.14_capi.rst
```

**Comment by hugovk** [comment]
> TODO: This was meant to be for the Python deprecations, not C API ones. I'll update it.

Updated.

File: Doc/deprecations/pending-removal-in-future.rst [hunk 1]
```diff
@@ -0,0 +1,147 @@
+Pending Removal in Future Versions
+----------------------------------
+
+The following APIs were will be removed in the future,
```
Comment (encukou) [inline Doc/deprecations/pending-removal-in-future.rst]:
```suggestion
The following APIs will be removed in the future,
```

**Review by encukou — APPROVED** [review]
Looks good to me, with one typo fix!

File: Doc/whatsnew/3.13.rst [hunk 2]
```diff
@@ -1631,318 +1631,13 @@ New Deprecations
 
 .. Add deprecations above alphabetically, not here at the end.
```
Comment (ezio-melotti) [inline Doc/whatsnew/3.13.rst:1637]: Would it make sense to add a new "Pending removal" section, with a paragraph describing what "pending removal" means, and include the other files as subsections?

These are currently nested under "New Deprecation" (not sure if intentionally):
image

I also noticed that there are two "Pending Removal in Python 3.x", for each "x", e.g.:
* https://cpython-previews--121241.org.readthedocs.build/en/121241/whatsnew/3.13.html#pending-removal-in-python-3-14
* https://cpython-previews--121241.org.readthedocs.build/en/121241/whatsnew/3.13.html#id7

Is this intentional?

Comment (encukou) [reply Doc/whatsnew/3.13.rst:1637]: I suggest not expanding the scope of the PR. Move the content in this one, rearrange the page later (and possibly only for 3.14).

> I also noticed that there are two "Pending Removal in Python 3.x", for each "x", e.g.:

Yes, there are separate sections for Python API and C API, and they're not presented very clearly.

File: Doc/deprecations/pending-removal-in-future.rst (hunk 1 above)
Comment (hugovk) [reply Doc/deprecations/pending-removal-in-future.rst]: Thanks, fixed at the same time as (partially...) resolving the conflict.

**Comment by hugovk** [comment]
I've opened https://github.com/python/cpython/issues/122085 as a tracking issue, and https://github.com/python/cpython/pull/122093 is the next PR.

File: Doc/whatsnew/3.13.rst (hunk 2 above)
Comment (ezio-melotti) [reply Doc/whatsnew/3.13.rst:1637]: FWIW, I followed up on this in the original issue:
* https://github.com/python/cpython/issues/122085#issuecomment-2258283199

**Pull request merged**